provisioning/transport/amqp/src/ProvisioningTransportHandlerAmqp.cs

```diff
@@ -191,66 +196,88 @@ private static async Task CreateLinksAsync(AmqpClientConnection connection, stri
-                    TimeoutConstant).ConfigureAwait(false);
-            ValidateOutcome(outcome);
-            var amqpResponse = await client.AmqpSession.ReceivingLink.ReceiveMessageAsync(TimeoutConstant)
-                .ConfigureAwait(false);
-            client.AmqpSession.ReceivingLink.AcceptMessage(amqpResponse);
-            string jsonResponse = await new StreamReader(amqpResponse.BodyStream).ReadToEndAsync()
-                .ConfigureAwait(false);
-            RegistrationOperationStatus status = JsonConvert.DeserializeObject<RegistrationOperationStatus>(jsonResponse);
-            status.RetryAfter = ProvisioningErrorDetailsAmqp.GetRetryAfterFromApplicationProperties(amqpResponse, DefaultOperationPoolingInterval);
-            return status;
+            using (var amqpMessage = AmqpMessage.Create(new AmqpValue { Value = DeviceOperations.GetOperationStatus }))
+            {
+                amqpMessage.Properties.CorrelationId = correlationId;
+                amqpMessage.ApplicationProperties.Map[MessageApplicationPropertyNames.OperationType] =
+                    DeviceOperations.GetOperationStatus;
+                amqpMessage.ApplicationProperties.Map[MessageApplicationPropertyNames.OperationId] = operationId;
+                var outcome = await client.AmqpSession.SendingLink
+                    .SendMessageAsync(
+                        amqpMessage,
+                        new ArraySegment<byte>(Guid.NewGuid().ToByteArray()),
+                        TimeoutConstant)
+                    .ConfigureAwait(false);
+                ValidateOutcome(outcome);
+                AmqpMessage amqpResponse = await client.AmqpSession.ReceivingLink.ReceiveMessageAsync(TimeoutConstant)
+                    .ConfigureAwait(false);
+                client.AmqpSession.ReceivingLink.AcceptMessage(amqpResponse);
+
+                using (var streamReader = new StreamReader(amqpResponse.BodyStream))
+                {
+                    string jsonResponse = await streamReader.ReadToEndAsync().ConfigureAwait(false);
+                    RegistrationOperationStatus status = JsonConvert.DeserializeObject<RegistrationOperationStatus>(jsonResponse);
+                    status.RetryAfter = ProvisioningErrorDetailsAmqp.GetRetryAfterFromApplicationProperties(amqpResponse, DefaultOperationPoolingInterval);
+                    return status;
+                }
+            }
         }
 
         private static DeviceRegistrationResult ConvertToProvisioningRegistrationResult(
             Models.DeviceRegistrationResult result)
         {
-            var status = ProvisioningRegistrationStatusType.Failed;
-            Enum.TryParse(result.Status, true, out status);
-
-            var substatus = ProvisioningRegistrationSubstatusType.InitialAssignment;
-            Enum.TryParse(result.Substatus, true, out substatus);
+            Enum.TryParse(result.Status, true, out ProvisioningRegistrationStatusType status);
```

Comment:
Default is Unassigned so will this change the behavior now?

Reply:
The .net contract for `out` parameters is that they must be assigned to. So our previous assignment of Failed didn't matter.

Does that answer your question?